### PR TITLE
Implement OPHASH160

### PR DESCRIPTION
**Pull request by supreme2580**
Closes: https://github.com/zig-bitcoin/btczee/issues/71

### CODE REVIEW

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: src/script/engine.zig [hunk 1]
```diff
@@ -453,11 +454,16 @@ pub const Engine = struct {
...
-        _ = try self.stack.pop();
-        // For now, just set the hash to a dummy value
-        const hash: [20]u8 = [_]u8{0x00} ** 20;
-        // TODO: Implement SHA256 and RIPEMD160
-        try self.stack.pushByteArray(&hash);
+        const data = try self.stack.pop();
+        defer self.allocator.free(data);
+
+        var hash: [Sha256.digest_length]u8 = undefined;
+        Sha256.hash(data, &hash, .{});
```
[the remark this answers is not in the document]
Comment (supreme2580) [reply src/script/engine.zig:461]: Ohhh sorry... lol will remove it in a sec

[the remark this answers is not in the document]
Comment (supreme2580) [reply src/script/engine.zig:461]: @tdelabro I just read the btcd implementation of opHash160 again it first uses the sha256 hash then get the ripemd160 hash of the sha256 hash, bitcoin primitives does the same thing, what'd you think

[the remark this answers is not in the document]
Comment (tdelabro) [reply src/script/engine.zig:461]: oooh okay you are right mb

**Review by tdelabro — APPROVED** [review]

File: src/script/engine.zig (hunk 1 above)
[the remark this answers is not in the document]
Comment (supreme2580) [reply src/script/engine.zig:461]: Okay boss

**Pull request merged**